File: Examples/CSharp/Articles/GetSetClassIdentifierEmbedOleObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Aspose.Cells.Drawing;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class GetSetClassIdentifierEmbedOleObject
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            // Load your sample workbook which contains embedded PowerPoint ole object
            Workbook wb = new Workbook(sourceDir + "sampleGetSetClassIdentifierEmbedOleObject.xls");

            // Access its first worksheet
            Worksheet ws = wb.Worksheets[0];

            // Access first ole object inside the worksheet
            OleObject oleObj = ws.OleObjects[0];

            // Convert 16-bytes array into GUID
            Guid guid = new Guid(oleObj.ClassIdentifier);

            // Print the GUID
            Console.WriteLine(guid.ToString().ToUpper());

            Console.WriteLine("GetSetClassIdentifierEmbedOleObject executed successfully.");
        }
    }
}

File: Examples/VisualBasic/Articles/GetSetClassIdentifierEmbedOleObject.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Aspose.Cells.Drawing

Namespace Articles
    Public Class GetSetClassIdentifierEmbedOleObject
        Public Shared Sub Run()
            ' ExStart:GetSetClassIdentifierEmbedOleObject
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Load your sample workbook which contains embedded PowerPoint ole object
            Dim wb As New Workbook(dataDir & Convert.ToString("sample.xls"))

            ' Access its first worksheet
            Dim ws As Worksheet = wb.Worksheets(0)

            ' Access first ole object inside the worksheet
            Dim oleObj As OleObject = ws.OleObjects(0)

            ' Convert 16-bytes array into GUID
            Dim guid As New Guid(oleObj.ClassIdentifier)

            ' Print the GUID
            Console.WriteLine(guid.ToString().ToUpper())
            ' ExEnd:GetSetClassIdentifierEmbedOleObject
        End Sub
    End Class
End Namespace
